schunk_sdh/ros/src/sdh.cpp

```diff
@@ -605,6 +609,62 @@ class SdhNode
   }
 
   /*!
+   * \brief Executes the service callback for emergency_stop.
+   *
+   * Performs an emergency stop.
+   * \param req Service request
+   * \param res Service response
+   */
+  bool srvCallback_EmergencyStop(std_srvs::Trigger::Request &req, std_srvs::Trigger::Response &res) {
+      try {
+        isInitialized_ = false;
+        sdh_->EmergencyStop();
+        sdh_->SetAxisEnable(sdh_->All, 0.0);
+        sdh_->SetAxisMotorCurrent(sdh_->All, 0.0);
+      }
+      catch(const SDH::cSDHLibraryException* e) {
+          ROS_ERROR("An exception was caught: %s", e->what());
+          res.success = false;
+          res.message = e->what();
+          return false;
```

Comment:
service callback should always `return true;` - even of the result of the service is not succesful.

`return false;` rather indicates that the call of the client could not connect with the server...